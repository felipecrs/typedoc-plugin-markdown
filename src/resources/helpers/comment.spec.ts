import * as Handlebars from 'handlebars';

import { setup } from '../../../test/utils';

describe(`[helpers] comment`, () => {
  let app: any;
  let project: any;
  let helpersComponent: any;

  beforeAll(() => {
    ({ app, project } = setup());
    helpersComponent = app.renderer.getComponent('helpers');
  });

  test(`should convert symbols brackets to symbol links'`, () => {
    expect(
      Handlebars.helpers.comment.call(project.findReflectionByName('commentsWithSymbolLinks').comment.text),
    ).toMatchSnapshot();
  });

  test(`should set warnings if symbol not found'`, () => {
    expect(helpersComponent.warnings.length > 0).toBeTruthy();
  });

  test(`should convert comments with includes'`, () => {
    expect(
      Handlebars.helpers.comment.call(project.findReflectionByName('commentsWithIncludes').comment.text),
    ).toMatchSnapshot();
  });

  test(`should build @link references'`, () => {
    expect(
      Handlebars.helpers.comment.call(project.findReflectionByName('functionWithDocLink').signatures[0].comment.text),
    ).toMatchSnapshot();
  });
});
