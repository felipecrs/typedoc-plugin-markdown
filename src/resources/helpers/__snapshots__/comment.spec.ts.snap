// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[helpers] comment should build @link references' 1`] = `
"Taken from http://usejsdoc.org/tags-inline-link.html.
"
`;

exports[`[helpers] comment should convert comments with includes' 1`] = `
"[[include:class-example.md]] [[include:VOID]]

This is an example of handlebars include

[[include:class-example.hbs]]
"
`;

exports[`[helpers] comment should convert symbols brackets to symbol links' 1`] = `
"- Link to an external reflection: [BaseClass](../classes/_classes_.baseclass.md)
- Link to an internal reflection: [commentsInReturn](_comments_.md#commentsinreturn)
- Link to an undefined reflection: [[VOID]]
"
`;
